File: test_login.js
const email = 'admin@assetflow.com';
const password = 'password123';

const formData = new URLSearchParams();
formData.append('username', email); // OAuth2 requires 'username'
formData.append('password', password);

async function testLogin() {
    console.log(`Attempting to log in as: ${email}`);
    try {
        const response = await fetch('http://localhost:8000/api/v1/identity/auth/login', {
            method: 'POST',
            headers: {
                'Content-Type': 'application/x-www-form-urlencoded',
            },
            body: formData.toString()
        });

        if (!response.ok) {
            const errData = await response.json();
            throw new Error(errData.detail || 'Login failed');
        }

        const data = await response.json();
        console.log("✅ SUCCESS! Login form returned valid token.");
        console.log(`Access Token: ${data.access_token.substring(0, 30)}...`);
        console.log(`Logged in User: ${data.user.first_name} ${data.user.last_name}`);
    } catch (err) {
        console.error("❌ FAILED:", err.message);
    }
}

testLogin();

File: test_login.py
import urllib.request
import json

url = 'http://127.0.0.1:8000/api/v1/identity/auth/login'
data = json.dumps({'email': 'admin@assetflow.com', 'password': 'password123'}).encode('utf-8')
req = urllib.request.Request(url, data=data)
req.add_header('Content-Type', 'application/json')

print(f"Attempting to log in to: {url}")
try:
    with urllib.request.urlopen(req) as response:
        result = json.loads(response.read().decode('utf-8'))
        print("✅ SUCCESS! Login form accepted the credentials.")
        print(f"Access Token: {result['access_token'][:30]}...")
        print(f"Logged in User: {result['user']['first_name']} {result['user']['last_name']}")
except urllib.error.HTTPError as e:
    err_data = json.loads(e.read().decode('utf-8'))
    print(f"❌ FAILED: {err_data.get('detail', 'Unknown error')}")
except Exception as e:
    print(f"❌ CRITICAL FAILURE: {e}")
